### PR TITLE
Course details page

### PR DESCRIPTION
**Pull request by YuriyMamoshin**
Some changes added to course details page

### CODE REVIEW

File: src/pages/course-details-page/components/main-content/components/reviews-bar/components/reviews-list/components/review/components/review-main/index.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+
+import styles from "./styles.module.scss";
+
+type ReviewMainProperties = {
+	text: string;
+};
+
+const ReviewMain: React.FC<ReviewMainProperties> = ({ text }) => {
```
Comment (KirillSerg) [inline src/pages/course-details-page/components/main-content/components/reviews-bar/components/reviews-list/components/review/components/review-main/index.tsx:12]: Is this component too small to separate? There is no separate logic here

File: src/pages/course-details-page/components/main-content/components/reviews-bar/components/reviews-stats/index.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	stats: ReviewsStats;
+};
+
+const ReviewsStatsBar: React.FC<ReviewsStatsProperties> = ({ stats }) => {
+	const [isRatingBarShown, setIsRatingBarShown] = useState<boolean>(false);
+
+	const { data: course } = useGetCourseByIdQuery("1");
+
+	const updateRatingDisplay = () => {
+		const screenWidth = window.innerWidth;
```
Comment (KirillSerg) [inline src/pages/course-details-page/components/main-content/components/reviews-bar/components/reviews-stats/index.tsx:23]: We have halper "getScreenWidth"

File: src/pages/course-details-page/components/main-content/components/categories-list/index.tsx [hunk 3]
```diff
@@ -1,13 +1,24 @@
...
 	return (
 		<div>
-			<h3 className={styles["categories_header"]}>Категорії курса</h3>
-			<Category categoryName="Інформаційні технології, IT" />
+			<h3 className={styles["categories__header"]}>Категорії курса</h3>
+			<article className={styles["categories__list"]}>
+				{course &&
+					course.categories.map((category, index) => {
+						return <CategoryBlock category={category} key={index} />;
+					})}
```
Comment (ValentinaTenus) [inline src/pages/course-details-page/components/main-content/components/categories-list/index.tsx:20]: You can use without return with ()

File: src/pages/course-details-page/components/main-content/components/reviews-bar/index.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+const ReviewsBar: React.FC = () => {
+	const { data: stats } = useGetReviewsStatsQuery(MOCK_COURSE_ID);
+
+	return (
+		<div className={styles["reviews-bar"]}>
+			<h3 className={styles["reviews-bar__header"]}>Відгуки</h3>
+			<ReviewsList />
+			{stats && <ReviewsStatsBar stats={stats} />}
+			<Button
```
Comment (KirillSerg) [inline src/pages/course-details-page/components/main-content/components/reviews-bar/index.tsx:21]: will there be an on-click here?

File: src/pages/course-details-page/components/main-content/components/navigation-bar/index.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+	return (
+		<nav className={styles["navbar__wrapper"]}>
+			<ul className={styles["navbar"]}>
+				<li
+					className={`${styles["navbar__item"]} ${active === "Про курс" ? styles["item_active"] : ""}`}
+					onClick={() => {
+						setActive("Про курс");
+						scrollToSection(aboutCourse);
+					}}
```
Comment (ValentinaTenus) [inline src/pages/course-details-page/components/main-content/components/navigation-bar/index.tsx:27]: Create handler for this

File: src/pages/course-details-page/index.tsx [hunk 6]
```diff
@@ -9,7 +9,7 @@ const CourseDetailsPage: React.FC = () => {
 	return (
 		<div className={styles["course_details_page"]}>
 			<Header />
-			<MainContent />
+			<MainContent courseId="125" />
```
Comment (KirillSerg) [inline src/pages/course-details-page/index.tsx]: it is temporary mock data?

File: src/pages/course-details-page/components/main-content/components/navigation-bar/index.tsx [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+				>
+					Про курс
+				</li>
+				<li
+					className={`${styles["navbar__item"]} ${active === "Про компанію" ? styles["item_active"] : ""}`}
+					onClick={() => {
+						setActive("Про компанію");
+						scrollToSection(aboutCompany);
+					}}
+				>
```
Comment (ValentinaTenus) [inline src/pages/course-details-page/components/main-content/components/navigation-bar/index.tsx:37]: I think it'll better define this in handler

**Review by KirillSerg — APPROVED** [review]

File: src/pages/course-details-page/components/main-content/components/reviews-bar/components/reviews-list/components/review/components/review-main/index.tsx [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+			<section className={styles["review__text"]} ref={textRef}>
+				{text}
+			</section>
+			{(isTextOverFlown || isFullTextShown) && (
+				<Button
+					className={styles["review__button"]}
+					onClick={() => {
+						setIsFullTextShown(!isFullTextShown);
+						defineTextHeight();
+					}}
```
Comment (ValentinaTenus) [inline src/pages/course-details-page/components/main-content/components/reviews-bar/components/reviews-list/components/review/components/review-main/index.tsx:45]: Consider using handler function

File: src/pages/course-details-page/components/main-content/components/reviews-bar/components/reviews-list/index.tsx [hunk 9]
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+
+import styles from "./styles.module.scss";
+// import { useGetCourseReviewsQuery } from "~/redux/reviews/reviews-course-api";
```
Comment (ValentinaTenus) [inline src/pages/course-details-page/components/main-content/components/reviews-bar/components/reviews-list/index.tsx:4]: Delete comment

File: src/pages/course-details-page/components/main-content/components/reviews-bar/components/reviews-list/index.tsx [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+import { Review } from "./components/review";
+
+// const mockCourseId = 7;
+
+const ReviewsList: React.FC = () => {
+	// const { data: reviews } = useGetCourseReviewsQuery(mockCourseId);
+	// console.log(reviews);
+
```
Comment (ValentinaTenus) [inline src/pages/course-details-page/components/main-content/components/reviews-bar/components/reviews-list/index.tsx:12]: Delete comments

File: src/pages/course-details-page/components/main-content/components/reviews-bar/components/reviews-list/components/review/components/review-main/index.tsx (hunk 1 above)
Comment (YuriyMamoshin) [reply src/pages/course-details-page/components/main-content/components/reviews-bar/components/reviews-list/components/review/components/review-main/index.tsx:12]: Yes, it didn't have, but now it does have some separate logic - a button that shows/hides more text.

File: src/pages/course-details-page/components/main-content/components/reviews-bar/components/reviews-stats/index.tsx (hunk 2 above)
Comment (YuriyMamoshin) [reply src/pages/course-details-page/components/main-content/components/reviews-bar/components/reviews-stats/index.tsx:23]: Thanks, added this hook

File: src/pages/course-details-page/components/main-content/components/reviews-bar/index.tsx (hunk 4 above)
Comment (YuriyMamoshin) [reply src/pages/course-details-page/components/main-content/components/reviews-bar/index.tsx:21]: Yes, this part of the page is still in progress. How it's just botton with mock text.

File: src/pages/course-details-page/index.tsx (hunk 6 above)
Comment (YuriyMamoshin) [reply src/pages/course-details-page/index.tsx]: Yes, now I've set it explicitly.

File: src/pages/course-details-page/components/main-content/components/categories-list/index.tsx (hunk 3 above)
Comment (YuriyMamoshin) [reply src/pages/course-details-page/components/main-content/components/categories-list/index.tsx:20]: Thanks, refactored.

File: src/pages/course-details-page/components/main-content/components/navigation-bar/index.tsx (hunk 5 above)
Comment (YuriyMamoshin) [reply src/pages/course-details-page/components/main-content/components/navigation-bar/index.tsx:27]: Thanks, done.

File: src/pages/course-details-page/components/main-content/components/navigation-bar/index.tsx (hunk 7 above)
Comment (YuriyMamoshin) [reply src/pages/course-details-page/components/main-content/components/navigation-bar/index.tsx:37]: Thanks. done

File: src/pages/course-details-page/components/main-content/components/reviews-bar/components/reviews-list/components/review/components/review-main/index.tsx (hunk 8 above)
Comment (YuriyMamoshin) [reply src/pages/course-details-page/components/main-content/components/reviews-bar/components/reviews-list/components/review/components/review-main/index.tsx:45]: Done, added a handler

File: src/pages/course-details-page/components/main-content/components/reviews-bar/components/reviews-list/index.tsx (hunk 9 above)
Comment (YuriyMamoshin) [reply src/pages/course-details-page/components/main-content/components/reviews-bar/components/reviews-list/index.tsx:4]: Done

File: src/pages/course-details-page/components/main-content/components/reviews-bar/components/reviews-list/index.tsx (hunk 10 above)
Comment (YuriyMamoshin) [reply src/pages/course-details-page/components/main-content/components/reviews-bar/components/reviews-list/index.tsx:12]: Done

**Review by ValentinaTenus — APPROVED** [review]

**Pull request merged**